src/test/java/org/jenkinsci/plugins/github_branch_source/EventsTest.java

```diff
@@ -188,7 +191,11 @@ private GHSubscriberEvent createEvent(String eventPayloadFile) throws IOExceptio
   private void waitAndAssertReceived(boolean received) throws InterruptedException {
     long watermark = SCMEvents.getWatermark();
     // event will be fired by subscriber at some point
-    SCMEvents.awaitOne(watermark, 1200, TimeUnit.MILLISECONDS);
+    SCMEvents.awaitOne(watermark, received ? 20 : 200, TimeUnit.MILLISECONDS);
+
+    if (received) {
+      TestSCMEventListener.awaitUntilReceived();
```

Comment:
usually event comes immediately, but if system struggling from not enough resources it could be longer, so let's add a waiter for event to be received for tests which are expecting `received = true`